src/youtube-player/youtube-player.ts

```diff
@@ -556,3 +586,37 @@ export class YouTubePlayer implements AfterViewInit, OnChanges, OnDestroy {
     );
   }
 }
+
+let apiLoaded = false;
+
+/** Loads the YouTube API from a specified URL only once. */
+function loadApi(): void {
+  if (apiLoaded) {
+    return;
+  }
+
+  // We can use `document` directly here, because this logic doesn't run outside the browser.
+  const url = 'https://www.youtube.com/iframe_api';
+  const script = document.createElement('script');
+  const callback = (event: Event) => {
+    script.removeEventListener('load', callback);
+    script.removeEventListener('error', callback);
+
+    if (event.type === 'error') {
+      apiLoaded = false;
+
+      if (typeof ngDevMode === 'undefined' || ngDevMode) {
+        console.error(`Failed to load YouTube API from ${url}`);
+      }
+    }
+  };
+  script.addEventListener('load', callback);
+  script.addEventListener('error', callback);
+  (script as any).src = url;
```

Comment:
The `as any` here is to keep the security test happy. This isn't an XSS risk since we always load `https://www.youtube.com/iframe_api`.